wordpress-mobile/WordPress-Android
# Issue 15828 - Notification comment support in the new comment editor

Fixes #15828

Currently only Site Comments are supported in the new comment editor. The goal of this PR is to also allow Notification Comments to be shown and editable in the new comment editor (instead of the legacy editor). It also tries to make it easier to support other Comment types in the future (e.g. Reader).

<p float="left">
<img width="400" alt="Screen Shot 2022-02-03 at 2 43 47 PM" src="https://user-images.githubusercontent.com/14964993/152401230-c3a2bd88-d59a-41e4-b6af-b2db0f132ce7.png">
<img width="400" alt="Screen Shot 2022-02-03 at 2 43 52 PM" src="https://user-images.githubusercontent.com/14964993/152401253-4f2f9531-cb72-4cc1-9f00-468534db8398.png">
</p>

<p float="left">
<img width="400" alt="Screen Shot 2022-02-03 at 2 43 59 PM" src="https://user-images.githubusercontent.com/14964993/152401259-85a8e2e6-6540-46a8-a648-dcecc74ea6e5.png">
<img width="400" alt="Screen Shot 2022-02-03 at 2 44 02 PM" src="https://user-images.githubusercontent.com/14964993/152401266-b4c16820-5368-4209-82c2-2376d48b9761.png">
</p>

To test:
1 - Login with an user that has at least one notification of a comment;
2 - Select any site;
3 - Tap on "Notifications" bottom navigation icon to open Notifications screen;
4 - Tap on "Comments" tab inside Notifications screen;
5 - Tap on the comment you want to test;
6 - Tap on "More" button;
7 - Tap on "Edit" item. The notification comment edit should be opened in the new comment editor;
8 - Edit the comment text;
9 - Tap on "Done";
10 - The comment text should be updated.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
-

3. What automated tests I added (or what prevented me from doing so)
- `CommentEssentialsTest.kt`
- `CommentSourceTest.kt`
- `CommentEssentialsExtensionKtTest.kt`
- `GetCommentUseCaseTest.kt`
- Included more tests in `UnifiedCommentsEditViewModelTest.kt`

PR submission checklist:
